build.go

```diff
@@ -79,6 +82,24 @@ func Build(entryResolver EntryResolver,
 			return packit.BuildResult{}, err
 		}
 
+		logger.Process("Checking custom npm version")
```

Comment:
I think we should move this into the if found, and log something like "Using a custom npm versions set with BP_NPM_VERSION"

That would reduce the updates needed to the tests, both in this repo but more importantly in my mind possibly to tests which existing in the other buildpack repos.

Reply:
In principal, I find it a bit worrying that you should not add a log message because it can break tests. Those tests are way to fragile anyway. But you have a point and probably the log message is better suited in the if anyway.